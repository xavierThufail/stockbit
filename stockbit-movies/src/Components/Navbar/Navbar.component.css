.ant-menu-item:active, .ant-menu-submenu-title:active {
  background: none;
}

.ant-menu-inline .ant-menu-item::after {
  border-right: 3px solid #c33e32;
}

.ant-menu-item-selected {
  color: white;
}

.ant-menu-item-selected span:first-of-type {
  color: #c33e32;
}

.ant-menu:not(.ant-menu-horizontal) .ant-menu-item-selected {
  background: none;
}

.ant-menu-item:hover {
  color: white;
}

.ant-menu-item:hover span:first-of-type {
  color: #c33e32;
}

.ant-menu-horizontal:not(.ant-menu-dark) > .ant-menu-item:hover, .ant-menu-horizontal:not(.ant-menu-dark) > .ant-menu-submenu:hover, .ant-menu-horizontal:not(.ant-menu-dark) > .ant-menu-item-active, .ant-menu-horizontal:not(.ant-menu-dark) > .ant-menu-submenu-active, .ant-menu-horizontal:not(.ant-menu-dark) > .ant-menu-item-open, .ant-menu-horizontal:not(.ant-menu-dark) > .ant-menu-submenu-open, .ant-menu-horizontal:not(.ant-menu-dark) > .ant-menu-item-selected, .ant-menu-horizontal:not(.ant-menu-dark) > .ant-menu-submenu-selected {
  color: white;
  border-bottom: none;
}

.ant-menu-horizontal .ant-menu-item, .ant-menu-horizontal .ant-menu-submenu-title {
  transition: none
}
