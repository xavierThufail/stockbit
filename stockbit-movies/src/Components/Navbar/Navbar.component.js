import React from 'react';
import { Menu } from 'antd';
import * as IconTypes from '@ant-design/icons/';
import { useHistory } from 'react-router-dom';

import routePaths from '../../Constants/routePaths';
import './Navbar.component.css';
import styles from './Navbar.style.js';
import config from './Navbar.config';

const Title = ({title}) => (<p style={styles.title}>{title}</p>);

const groupMapping = (items, history) => {
  const titleGroups = [];

  for (const [key] of Object.entries(items)) {
    titleGroups.push(key);
  };

  return titleGroups.map((key, index) => {
    let title = key.split('');
    title[0] = title[0].toUpperCase();
    title = title.join('');

    return (
      <Menu.ItemGroup title={<Title title={title} />} key={index} >
        {items[key].map(itemMapping({}, history))}
      </Menu.ItemGroup>
    )}
  )
};

const mappingHandleClick = ({ title }, history) => {
  return title === 'Home'
    ? () => history.push(routePaths.DASHBOARD)
    : () => {console.log('Hahahahaha')};
}

const itemMapping = (style, history) => (item, index) => {
  const Icon = IconTypes[item.icon];
  
  return (
    <Menu.Item onClick={mappingHandleClick(item, history)} key={item.key} style={style} icon={item.icon ? (<Icon />) : (<div />) }>
      {item.title}
    </Menu.Item>
  )
};

const Navbar = ({ items, styleNavbar, styleItem, mode, onClick, subMenu }) => {
  const history = useHistory();

  return (
    <Menu style={ styleNavbar } mode={mode} onClick={onClick} >
      {subMenu
      ? groupMapping(items, history)
      : items.map(itemMapping(styleItem))}
    </Menu>
  );
};

Navbar.displayName = config.displayName;
Navbar.propTypes = config.propTypes;

export default Navbar;