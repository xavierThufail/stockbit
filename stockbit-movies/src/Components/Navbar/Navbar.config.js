import PropTypes from 'prop-types';

const propTypes = {
  items: PropTypes.array.isRequired,
  styleNavbar: PropTypes.object,
  styleItem: PropTypes.object,
  mode: PropTypes.string,
  onClick: PropTypes.func,
  subMenu: PropTypes.bool
};

const displayName = 'Navbar';

const config = {
  displayName,
  propTypes
};

export default config;
