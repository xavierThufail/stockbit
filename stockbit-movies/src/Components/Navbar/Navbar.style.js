const styles = {
  title: {
    fontWeight: 'bold',
    color: '#8f8f8f',
    marginTop: '30px',
    marginBottom: '0px'
  }
};

export default styles;
