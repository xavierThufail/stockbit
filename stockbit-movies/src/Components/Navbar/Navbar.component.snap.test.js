import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Navbar from './Navbar.component';
import constants from '../../Constants/navbar';

jest
  .mock('react-router-dom', () => ({
    Redirect: () => 'Redirect',
    useHistory: jest.fn()
  }))

describe('Navbar component snap test', () => {
  it('Render Navbar, should match snapshot', () => {

    const { container } = render(<Navbar items={constants.dashboard} />);

    expect(container).toMatchSnapshot();
  });
});
