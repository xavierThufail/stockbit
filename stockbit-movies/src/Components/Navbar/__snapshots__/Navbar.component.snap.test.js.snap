// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component snap test Render Navbar, should match snapshot 1`] = `
<div>
  <ul
    class="ant-menu ant-menu-light ant-menu-root ant-menu-vertical"
    role="menu"
  >
    <li
      class="ant-menu-item"
      role="menuitem"
    >
      <div
        class="ant-menu-item-icon"
      />
      <span>
        Tv Series
      </span>
    </li>
    <li
      class="ant-menu-item"
      role="menuitem"
    >
      <div
        class="ant-menu-item-icon"
      />
      <span>
        Movies
      </span>
    </li>
    <li
      class="ant-menu-item"
      role="menuitem"
    >
      <div
        class="ant-menu-item-icon"
      />
      <span>
        Animes
      </span>
    </li>
  </ul>
</div>
`;
